client/state/posts/selectors.js

```diff
@@ -287,19 +287,22 @@ export function isRequestingSitePost( state, siteId, postId ) {
  * @param  {Number} postId Post ID
  * @return {Object}        Post object with revisions
  */
-export function getEditedPost( state, siteId, postId ) {
-	const post = getSitePost( state, siteId, postId );
-	const edits = getPostEdits( state, siteId, postId );
-	if ( ! edits ) {
-		return post;
-	}
+export const getEditedPost = createSelector(
+	( state, siteId, postId ) => {
+		const post = getSitePost( state, siteId, postId );
+		const edits = getPostEdits( state, siteId, postId );
+		if ( ! edits ) {
+			return post;
+		}
 
-	if ( ! post ) {
-		return edits;
-	}
+		if ( ! post ) {
+			return edits;
+		}
```

Comment:
is this right? we potentially return `post` and at a further location we perform a check if it actually exists… are we asserting that one of the two objects will always exist? what if neither exists?